### PR TITLE
Wrote Upload methods

### PR DESCRIPTION
**Pull request by MMDPROJECT**
I have only wrote `upload` methods.
I will write download methods for clientSide while linking backEnd to frontEnd
@Retriever5500 @Navid-Ebadi-2003

### CODE REVIEW

**Comment by Retriever5500** [comment]
Okay looks good, but seems like there is some repeated code. Will show them in review.
@MMDPROJECT

File: src/main/java/Server/Service.java [hunk 1]
```diff
@@ -140,4 +140,238 @@ public void executeRequest(JsonObject jsonRequest) {
...
+    // Upload method to send "goHomePage" Results from the server to the client
+    public void uploadProfilePics(JsonArray jsonArray) throws IOException {
+        OutputStream outputStream = serverSocket.getOutputStream();
+        DataOutputStream dataOutputStream = new DataOutputStream(outputStream);
+
+        for (int i = 0; i < jsonArray.size(); i++) {
+            // Extract profilePath
+            JsonObject arrayItem = jsonArray.get(i).getAsJsonObject();
+            String profilePath = arrayItem.get("profilePath").getAsString();
+            File file = new File(profilePath);
```
Comment (Retriever5500) [inline src/main/java/Server/Service.java]: Here we don't need to write the code to send the file again, we have already written that in `uploadProfilePic`. We just have to replace that section with `uploadProfilePic(profilePath)`

File: src/main/java/Server/Service.java [hunk 2]
```diff
@@ -140,4 +140,238 @@ public void executeRequest(JsonObject jsonRequest) {
...
+        int bytesRead;
+        while ((bytesRead = fileInputStream.read(buffer)) != -1) {
+            outputStream.write(buffer, 0, bytesRead);
+        }
+
+        dataOutputStream.flush();
+        outputStream.flush();
+        fileInputStream.close();
+    }
+    public void uploadTracks(JsonArray jsonArray) throws IOException {
```
Comment (Retriever5500) [inline src/main/java/Server/Service.java]: What's the difference between `uploadProfilePics` and `uploadTracks`? Seems like their code is (almost) identical. I think it's better if we just have two methods: `uploadFile(String filePath)` and `uploadFiles(JsonArray fileArray, String filePathKey)`, where `filePathKey` is the JsonObject's key to the file path.

File: src/main/java/Server/Service.java [hunk 3]
```diff
@@ -140,4 +140,238 @@ public void executeRequest(JsonObject jsonRequest) {
...
+        // Template of jsonResult:
+        // {
+        // "userId" : %s,
+        // "username" : %s,
+        // "profilePath" : %s
+        // }
+
+        // Parsing Results
+        String profilePath = jsonResults.get("profilePath").getAsString();
+        JsonArray createdPlaylistsJson = jsonResults.getAsJsonArray("createdPlaylists");
```
Comment (Retriever5500) [inline src/main/java/Server/Service.java:232]: Here why aren't we sending the user's followers and followings profile pictures? We're just viewing their names?

File: src/main/java/Server/Service.java (hunk 1 above)
Comment (MMDPROJECT) [reply src/main/java/Server/Service.java]: Oh yes you are right. It will be fixed in a moment

File: src/main/java/Server/Service.java (hunk 2 above)
Comment (MMDPROJECT) [reply src/main/java/Server/Service.java]: Yes you are completely right and I apologize. I will get done exactly like you said

**Comment by MMDPROJECT** [comment]
Man I appreciate your attention and help in reviewing this pull request, I have made the changes you said but the last review, because I think I have to review other parts of the code. I try to get to that till tonight 
@Retriever5500

**Comment by Retriever5500** [comment]
It's alright, take your time

**Pull request merged**